cpp/include/cudf/detail/iterator.cuh

```diff
@@ -313,6 +313,72 @@ struct scalar_pair_accessor : public scalar_value_accessor<Element> {
   }
 };
 
+/**
+ * @brief Utility to fetch representation data type of specified Element type.
+ *
+ * By default, rep_type_impl<Element>::type == Element.
+ */
+template <typename Element, typename Ignore = void>
+struct rep_type_impl {
+  using type = Element;
+};
```

Comment:
I think `device_storage_type_t` is what you want to use here instead.

Reply:
I had started out considering that `device_storage_type_t` only differentiates between `fixed_point` and everything else. `timestamp` and `duration` also have a `T::rep`, which could be added here later.

Given that the focus of the PR is decimals, I'll switch to `device_storage_type_t`.